manifests.go

```diff
@@ -0,0 +1,43 @@
+package distribution
+
+import "github.com/docker/distribution/digest"
+
+// Manifest specifies a registry object with a list of dependencies.
+type Manifest interface {
+	Blob // io access to serialized content.
+
+	// Dependencies returns a list of object on which this manifest depends.
+	// Typically, these are layers.
+	Dependencies() []Descriptor
+}
+
+// ManifestBuilder creates a manifest allowing one to include dependencies.
+// Instances can be obtained from a version-specific manifest package, likely
+// from a type that provides a version specific interface.
+type ManifestBuilder interface {
+	// Build creates the manifest fromt his builder.
+	Build() Manifest
+
+	// AddDependency includes the dependency in the manifest. If the add
+	// fails, such as when adding an unsupported dependency, an error may be
+	// returned.
+	AddDependency(dependency Descriptor) error
+}
+
+// ManifestService describes operations on image manifests.
```

Comment:
It is not a one-off, special media type. There is semantic meaning to putting a manifest versus just putting a blob. The presence of dependencies is verified.

Please propose a different interface that keeps things simple and provides these semantics if you don't think this will work.